static/docs/commands-reference/commit.md

```diff
@@ -2,40 +2,212 @@
+  not happen.
+* Adds the file to the DVC cache
+
+There are many cases where the last step is not desirable. For the DVC commands
+where it is appropriate the `--no-commit` option prevents the last step from
+occurring. The checksum is still computed and added to the DVC file, but the
+file is not added to the cache.
+
+That's where the `dvc commit` command comes into play. It handles that last
+step of adding the file to the DVC cache.
+
+The `dvc commit` command is useful for several scenarios where a stage is in
+development, or takes a long time to run, or must be run on another machine
+with higher compute power. The key is to run DVC commands in a mode where data
+is not immediately committed to the cache (the `--no-commit` or `--no-exec`
+options), and to commit the data as a separate step performed when it is certain
+the data is finalized.
+
+* Code for a stage is under active development, with lots of temporary files
+  sitting in the workspace. Committing those temporary files would clutter up
+  the cache, so it's better to commit them when development is finished.
+* Execution of a stage takes a long time to run, like 24 hours, or requires
+  specialized hardware like GPU's or an HPC cluster. The pipeline and cache
+  might be on one machine, like a laptop, and execution of the long-running
+  stage might be manually coordinated with the data added to the cache using
+  `dvc commit` when ready.
+
 ## Options
 
+* `-d`, `--with-deps` - determines the files to commit by searching backwards in
+  the pipeline from the named stage(s). The only files which will be committed
+  are associated with the named stage, and the stages which execute earlier in
+  the pipeline.
+
+* `-R`, `--recursive` - the `targets` value is expected to be a directory path.
+  With this option, `dvc commit` determines the files to commit by searching the
+  named directory, and its subdirectories, for DVC files for which to commit
+  data. Along with providing a `target`, or `target` along with `--with-deps`,
+  it is yet another way to limit the scope of DVC files to upload.
+
+* `-f`, `--force` - commit data even if checksums for dependencies or outputs
+  did not change.
+
 * `-h`, `--help` - prints the usage/help message, and exit.
 
-* `-q`, `--quiet` - does not write anything to standard output. Exit with 0 if
-  no problems arise, otherwise 1.
+* `-q`, `--quiet` - do not write anything to standard output. Exit with 0 if all
+  stages are up to date or if all stages are successfully rerun, otherwise exit
+  with 1.
+
+* `-v`, `--verbose` - displays detailed tracing information from executing the
+  `dvc add` command.
 
-* `-v`, `--verbose` - displays detailed tracing information.
 
 ## Examples
 
+Let's start with the example workspace used in the
+[Getting Started](/doc/get-started) tutorial.  Setup requires these steps:
+
+```dvc
+    $ git clone https://github.com/iterative/example-get-started
+    $ cd example-get-started
+    $ dvc pull
```

Comment:
please, take a look at the checkout, I would just copy the "Click and expand to setup the project" section from it. It mentions virtualenv or something similar and pip install to install dependencies. This is required to be able to run the code.

Reply:
copied in